lib/pharos/kube/stack.rb

```diff
@@ -53,23 +52,29 @@ def apply
       def prune(checksum)
         pruned = []
 
-        @session.api_groups.each do |api_group|
-          group_client = @session.resource_client(api_group.preferredVersion.groupVersion)
+        @session.api_versions.each do |api_version|
+          client = @session.resource_client(api_version)
+          client.entities.each do |method_name, entity|
+            next if method_name.end_with?('_review')
+            next if api_version == 'v1' && entity.resource_name == 'bindings' # XXX: the entity definition does not have the list verb... but kubeclient does not expose that
+            next if api_version == 'v1' && entity.resource_name == 'componentstatuses' # apiserver ignores the ?labelSelector query
+            next if api_version == 'v1' && entity.resource_name == 'endpoints' # inherits stack labels from service, does not have any ownerReference...
```

Comment:
Only way I can think of avoiding having to hardcode this would be to skip resources that have an `ownerReference` that would cause them to get deleted later during the prune... but service endpoints don't have any `ownerReference`, they must get GC'd magically :(

Reply:
Maybe prune should not delete resources that do not have the `pharos.kontena.io/stack-checksum` annotation? At least not when pruning as part of a stack apply, maybe as part of an addon uninstall => full prune?